app/views/unattended/provisioning_templates/snippet/preseed_netplan_generic_interface.erb

```diff
@@ -17,24 +17,24 @@ oses:
         dhcp4: <%= @dhcp %>
         dhcp6: <%= @dhcp6 %>
 <%- if !@dhcp && !@subnet.nil? && !@interface.ip.nil? -%>
-        addresses: [ <%= @interface.ip %>/<%= @subnet.cidr %> ]
+        addresses:
+          - <%= @interface.ip %>/<%= @subnet.cidr %>
+<%- if !@dhcp6 && !@subnet6.nil? && !@interface.ip6.nil? -%>
+          - <%= @interface.ip6 %>/<%= @subnet6.cidr %>
+<%-   end -%>
 <%-   if @subnet.gateway.present? -%>
         gateway4: <%= @subnet.gateway %>
 <%-   end -%>
-<%-   if @interface.primary -%>
-        nameservers:
-          search: [ <%= @interface.domain %> ]
-          addresses: <%= @subnet.dns_servers %>
-<%-   end -%>
-<%- end -%>
 <%- if !@dhcp6 && !@subnet6.nil? && !@interface.ip6.nil? -%>
-        addresses: [ <%= @interface.ip6 %>/<%= @subnet6.cidr %> ]
 <%-   if @subnet6.gateway.present? -%>
-        gateway4: <%= @subnet6.gateway %>
+        gateway6: <%= @subnet6.gateway %>
 <%-   end -%>
+<%- end -%>
 <%-   if @interface.primary -%>
         nameservers:
           search: [ <%= @interface.domain %> ]
-          addresses: <%= @subnet6.dns_servers %>
-<%-   end -%>
+          addresses:
+<% @subnet.dns_servers.each do |dns_server| %>            - <%= dns_server %>
+<% end %><% @subnet6.dns_servers.each do |dns6_server| %>            - <%= dns6_server %>
```

Comment:
You can use `-%>` to control whitespace. I think this results in the same output, but is easier to read.
```suggestion
<%- end -%>
<%- @subnet6.dns_servers.each do |dns6_server| -%>
            - <%= dns6_server %>
```